From ae19b38d9a83238439d453d3aac4e2bae9bbfad1 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:25 -0400
Subject: [infiniband] usnic: Use GFP_ATOMIC under spinlock

Message-id: <ed6ea7391e07b15f3ec09f8edaf3861ec5d157fa.1411569661.git.dledford@redhat.com>
Patchwork-id: 96586
O-Subject: [Patch RHEL7 23/29] IB/usnic: Use GFP_ATOMIC under spinlock
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

This is called from qp_grp_and_vf_bind() and we are holding the
vf->lock so the allocation can't sleep.

Fixes: e3cf00d0a87f ('IB/usnic: Add Cisco VIC low-level hardware driver')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 8ce96afa8239f13bdf5ab35839bac46c103bbedc)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic_uiom.c b/drivers/infiniband/hw/usnic/usnic_uiom.c
index ae6934c..16755cd 100644
--- a/drivers/infiniband/hw/usnic/usnic_uiom.c
+++ b/drivers/infiniband/hw/usnic/usnic_uiom.c
@@ -498,7 +498,7 @@ int usnic_uiom_attach_dev_to_pd(struct usnic_uiom_pd *pd, struct device *dev)
  struct usnic_uiom_dev *uiom_dev;
  int err;
 
- uiom_dev = kzalloc(sizeof(*uiom_dev), GFP_KERNEL);
+ uiom_dev = kzalloc(sizeof(*uiom_dev), GFP_ATOMIC);
  if (!uiom_dev)
   return -ENOMEM;
  uiom_dev->dev = dev;
-- 
1.7.1